apps/server/src/modules/tool/service/validation/tool-validation.service.spec.ts

```diff
@@ -116,6 +118,7 @@ describe('ToolValidation', () => {
 			it('should return without error if matches', async () => {
 				const externalToolDO: ExternalToolDO = externalToolDOFactory.buildWithId();
 				externalToolDO.id = 'toolId';
+				externalToolService.isOauth2Config.mockReturnValue(false);
```

Comment:
I dont get this... the test is about the parameter id being the same as the tool id.
It used to check that it succeeds the update when nothing is changed.
But now you introduce that it doesnt have a valid oauthconfig, which makes the code skip the check in the firstplace, and always succeed without validating?

Im pretty sure this should be set to true here

Reply:
That should be the happy case of the validation. But i changed it now to oauth.
And created a new test for another config type like you suggested in your other comment